bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/discovery/HarmonyHubDiscoveryService.java

```diff
@@ -245,39 +247,30 @@ private void run() {
                             break;
                         }
                         logger.trace("READ {}", input);
-                        Properties properties = readProperties(input);
-
-                        String friendlyName = properties.getProperty("friendlyName");
-                        if (!responses.contains(friendlyName)) {
-                            responses.add(friendlyName);
-                            hubDiscovered(properties);
+                        // response format is key1:value1;key2:value2;key3:value3;
+                        Map<String, String> properties = Stream.of(input.split(";")).map(line -> line.split(":", 2))
+                                .collect(Collectors.toMap(entry -> entry[0], entry -> entry[1]));
+                        String friendlyName = properties.get("friendlyName");
+                        String hostName = properties.get("host_name");
+                        String ip = properties.get("ip");
+                        if (StringUtils.isNotBlank(friendlyName) && StringUtils.isNotBlank(hostName)
+                                && StringUtils.isNotBlank(ip) && !responses.contains(hostName)) {
+                            responses.add(hostName);
+                            hubDiscovered(ip, friendlyName, hostName);
                         }
                     }
-                } catch (IOException e) {
+                } catch (IOException | RuntimeException e) {
```

Comment:
Why catching RuntimeException? Is there no more specific exception type possible? If not, you should at least do debug logging in any case (and not just "if running").

Reply:
Since this is discovery and happens frequently, and we are at the mercy of what responds on a users network, this was a safe guard to catch exceptions parsing the response ( mainly index out of bounds on the Stream, which i can catch explicitly) 

As for logging, the "running" flag was put in early to avoid unnecessary logging when the discovery process was canceled/timeout and the server shut down ( i think, it's been a while) which would throw the a IOException.    So i would prefer to keep that.